core/math/geometry_2d.h

```diff
@@ -181,8 +181,7 @@ class Geometry2D {
 		D = Vector2(D.x * Bn.x + D.y * Bn.y, D.y * Bn.x - D.x * Bn.y);
 
 		// Fail if C x B and D x B have the same sign (segments don't intersect).
-		// (equivalent to condition (C.y < 0 && D.y < CMP_EPSILON) || (C.y > 0 && D.y > CMP_EPSILON))
-		if (C.y * D.y > CMP_EPSILON) {
```

Comment:
There might also be the possibility to use `CMP_EPSILON2`, i.e. the square of `CMP_EPSILON`, which is closer to 0. But the explicit check is also fine.

Reply:
I've tried to use `CMP_EPSILON2` and it didn't go well (some unit tests with points exactly on polygon edges were failing).

I thought multiplying the values and using just one threshold was equivalent to the full check, but it looks like it's not the case. It could be because of losing float precision while multiplying, I'm not sure.